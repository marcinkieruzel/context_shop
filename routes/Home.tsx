import React, { useState } from "react";
import { View, Text, Button } from "react-native";

type Props = {};

const Home: React.FC<Props> = ({}): JSX.Element => {

  // Dodatkowo przyk≈Çad testu
  const [text, setText] = useState("Hello");
  return (
    <View>
      <Text>Hej</Text>
      <Text testID="text">{text}</Text>
      <Button
        testID="press"
        title="Click"
        onPress={() => {
          setText("Bye");
        }}
      ></Button>
    </View>
  );
};

export default Home;
