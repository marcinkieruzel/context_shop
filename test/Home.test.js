jest.useFakeTimers();
import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";

import Home from "../routes/Home";

describe("<Home />", () => {

  it("match snapshot", () => {
      const snap = renderer.create(<Home />).toJSON()
      expect(snap).toMatchSnapshot()
  })


  it("renders", () => {
    const { getByText, getByTestId } = render(<Home />);

    getByText('Hello');

    expect(getByTestId('text').props.children).toBe("Hello")

    fireEvent.press(getByTestId("press"))

    expect(getByTestId('text').props.children).toBe("Bye")
  });
  it("has 3 child", () => {
    const tree = renderer.create(<Home />).toJSON();
    console.log(tree);
    expect(tree.children.length).toBe(3);
  });
});