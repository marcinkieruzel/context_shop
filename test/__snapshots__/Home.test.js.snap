// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> match snapshot 1`] = `
<View>
  <Text>
    Hej
  </Text>
  <Text
    testID="text"
  >
    Hello
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="press"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Click
      </Text>
    </View>
  </View>
</View>
`;
